services/data_service.py

```diff
@@ -31,13 +32,33 @@ def get_documentation(self, doc_id) -> FirestoreDoc | None:
         document_dict["id"] = document_snapshot.id
         firestore_doc = FirestoreDoc(**document_dict)
         return firestore_doc
+    
+    def get_user_documentation(self, user_id, doc_id) -> FirestoreDoc | None:
+        user_documentation_query = self._query(self.DOCUMENTATION_COLLECTION, [
```

Comment:
i wonder if it's worth it to do a query here. `doc_id` will always be unique in Firestore, therefore no two users should have the same `doc_id`.

Maybe we should `self._get` the doc, and then throw an unauthorized error if the owner id is not the same. That way we avoid querying